lib/spack/spack/database.py

```diff
@@ -309,7 +309,11 @@ def reindex(self, directory_layout):
                 for spec in directory_layout.all_specs():
                     # Create a spec for each known package and add it.
                     path = directory_layout.path_for_spec(spec)
-                    self._add(spec, path, directory_layout)
+                    old_info = old_data.get(spec.dag_hash())
+                    explicit = False
```

Comment:
`spack install <package that's already been installed>` sets the flag to true already, as an FYI.